### PR TITLE
Users routes

### PR DESCRIPTION
**Pull request by RainproDash**
handling of the users routes, also create a line into the user table when connected for the first

### CODE REVIEW

File: backend/src/entity/user.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    id: number;
+
+    @Column({ type: 'varchar', length: 255, nullable: false })
+    email: string;
+
+    @Column({ type: 'varchar', length: 255, nullable: false })
+    token: string;
+
+    @Column({ type: 'bigint', array: true, default: '{}' })
+    projects_id: number[];
```
Comment (alanschnegg) [inline backend/src/entity/user.ts]: This relations is already set in project entity, there is no use there

File: backend/src/entity/user.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Column({ type: 'boolean', nullable: false })
+    role: boolean;
```
Comment (alanschnegg) [inline backend/src/entity/user.ts]: Why is a 'role' in boolean ? Either rename this to 'superUser' or 'admin' or change its type with a relation table to a role entity

File: backend/src/routes/users/router_users.ts [hunk 3]
```diff
@@ -25,10 +27,98 @@ function isAuthenticated(req: Request, res: Response, next: NextFunction): void
...
+ * /users/:
+ *  get:
+ *     summary: Retrieve all users.
+ *     description: Endpoint to get a list of all users.
+ *     responses:
+ *       200:
+ *         description: Successful response with the list of users.
+ *       500:
+ *         description: Internal Server Error in case of any issues during the retrieval.
+ */
```
Comment (alanschnegg) [inline backend/src/routes/users/router_users.ts]: Where is the tag

File: backend/src/routes/users/router_users.ts [hunk 4]
```diff
@@ -25,10 +27,98 @@ function isAuthenticated(req: Request, res: Response, next: NextFunction): void
...
+ *         description: ID of the user to retrieve.
+ *         required: true
+ *         schema:
+ *           type: integer
+ *     responses:
+ *       200:
+ *         description: Successful response with the specified user.
+ *       500:
+ *         description: Internal Server Error in case of any issues during the retrieval.
+ */
```
Comment (alanschnegg) [inline backend/src/routes/users/router_users.ts]: Where is the tag

File: backend/src/routes/users/router_users.ts [hunk 5]
```diff
@@ -25,10 +27,98 @@ function isAuthenticated(req: Request, res: Response, next: NextFunction): void
...
+ *         application/json:
+ *           example:
+ *             projects_id: [1, 2, 3, 4]
+ *             role: true
+ *     responses:
+ *       200:
+ *         description: Successful response indicating the update of the specified user.
+ *       500:
+ *         description: Internal Server Error in case of any issues during the update.
+ */
```
Comment (alanschnegg) [inline backend/src/routes/users/router_users.ts]: Where is the tag

**Review by alanschnegg — CHANGES_REQUESTED** [review]

File: backend/src/entity/user.ts (hunk 1 above)
Comment (RainproDash) [reply backend/src/entity/user.ts]: I though it would be easier to parse, if we have a page where the user would be able to see his own projects

**Review by alanschnegg — APPROVED** [review]

**Pull request merged**